lib/xenops_server.ml

```diff
@@ -1400,13 +1401,12 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.t) : unit =
 			debug "VM.receive_memory %s power_state = %s" id (state.Vm.power_state |> rpc_of_power_state |> Jsonrpc.to_string);*)
 
 			(* set up the destination domain *)
-			perform_atomics (
-				simplify [VM_create (id, Some memory_limit);] @
-				(* Perform as many operations as possible on the destination domain before pausing the original domain *)
-				(atomics_of_operation (VM_restore_vifs id))
-			) t;
-
 			(try
+				perform_atomics (
```

Comment:
Why was this code moved here? I don't think it is needed to fix the issue that the patch intends to fix. Also, if `VM_create` or `VM_restore_vifs` now fail, a `Handshake.Error` will be sent, which is not right.